plugin/nfs/nfs.py

```diff
@@ -0,0 +1,496 @@
+                    efile.write("%s %s(%s)\n" % (exp._export_path, host, NFSPlugin._print_option(opts)))
+
+        except IOError:
+            raise LsmError(ErrorNumber.PLUGIN_BUG,
+                           'error writing exports')
+        finally:
+            efile.close()
+
+    @staticmethod
+    def _update_exports():
+        """Update the system"""
+        cmd = ["/usr/sbin/exportfs", "-ar"]
+        try:
+            subprocess.check_call(cmd)
+        except subprocess.CalledProcessError:
+            raise LsmError(ErrorNumber.INVALID_ARGUMENT,
+                           'exportfs failed')
+        except OSError:
+            raise LsmError(ErrorNumber.PLUGIN_BUG,
+                           'error calling exportfs')
+
+    def plugin_info(self, flags=0):
+        return "Local NFS Exports", VERSION
+
+    def __init__(self):
+        self.tmo = 0
+        return
+
+    def plugin_register(self, uri, password, timeout, flags=0):
+        self.tmo = timeout
+        if os.geteuid() != 0:
+            raise LsmError(ErrorNumber.INVALID_ARGUMENT,
+                           "This plugin requires root privilege for both " \
+                           "daemon and client")
+
+        return
+
+    def time_out_set(self, ms, flags=0):
+        self.tmo = ms
+
+    def time_out_get(self, flags=0):
+        return self.tmo
+
+    @staticmethod
+    def _get_fs_sizes(path):
+        """Stat a filesystem and return total and available space"""
+        try:
+            st = os.statvfs(path)
+            total_size = st.f_blocks * st.f_frsize
+            avail_size = st.f_bavail * st.f_frsize
+        except OSError:
+            raise
+
+        return (total_size, avail_size)    
+
+    def pools(self, search_key=None, search_value=None, flags=0):
+        pools = []
+        parts = list_mounts()
+        for p in parts:
+            try:
+                (total_size, avail_size) = NFSPlugin._get_fs_sizes(p)
+                fsid = get_fsid(p)
+                pooltype = Pool.ELEMENT_TYPE_FS
+                unsup_actions = 0
+                status = System.STATUS_OK
+                status_info = ''
+                pools.append(Pool(fsid, p, pooltype, unsup_actions,
+                                  total_size, avail_size, status, status_info,
+                                  self._SYSID))
+            except OSError:
+                pass
+        return search_property(pools, search_key, search_value)
+
+    def systems(self, flags=0):
+        syslist = []
+        hostname = os.uname()[1]
+        syslist.append(System(NFSPlugin._SYSID, hostname, System.STATUS_UNKNOWN, ''))
+        return syslist
+
+    def fs(self, search_key=None, search_value=None, flags=0):
+        """List filesystems, required for other operations"""
+        fss = []
+        parts = list_mounts()
+        for p in parts:
+            try:
+                (total_size, avail_size) = NFSPlugin._get_fs_sizes(p)
+                fsid = get_fsid(p)
```

Comment:
So are you saying that if there are no filesystems to list then that it is an error rather than an empty list ?